### PR TITLE
removing $(OS) uses from the libraries

### PR DESCRIPTION
**Pull request by Anipik**
Working towards https://github.com/dotnet/runtime/issues/32451

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@Anipik can you please share the motivation for doing this? Can you share your discussion with Eric?

**Comment by Anipik** [comment]
Earlier we didnt have a good way in msbuild to get the os we are running on. so we started using this shell property to get that.

We currently have 2 properties i.e. OSGroup and OS representing the same thing in our libraries subset. We can calculate the OSGroup without the need of the OS. We are already doing it for non-windows OSES https://github.com/dotnet/runtime/pull/32673/files#diff-d3bedba39ce6da3e37ca0db83d7ef035R30

File: src/libraries/Directory.Build.props [hunk 1]
```diff
@@ -29,7 +29,7 @@
     <DefaultOSGroup Condition="'$(DefaultOSGroup)' == '' and $([MSBuild]::IsOSPlatform('FREEBSD'))">FreeBSD</DefaultOSGroup>
     <DefaultOSGroup Condition="'$(DefaultOSGroup)' == '' and $([MSBuild]::IsOSPlatform('NETBSD'))">NetBSD</DefaultOSGroup>
     <DefaultOSGroup Condition="'$(DefaultOSGroup)' == '' and $([MSBuild]::IsOSUnixLike())">Linux</DefaultOSGroup>
-    <DefaultOSGroup Condition="'$(DefaultOSGroup)' == ''">$(OS)</DefaultOSGroup>
+    <DefaultOSGroup Condition="'$(DefaultOSGroup)' == '' and $([MSBuild]::IsOSPlatform('WINDOWS'))">Windows_NT</DefaultOSGroup>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.props]: Why not keep the `$(OS)` fallback here?

Comment (ViktorHofer) [reply src/libraries/Directory.Build.props]: Are you doing this to switch to the RID layout and not using `$(OS)` at all anymore?

**Comment by ViktorHofer** [comment]
Thanks, that makes sense. Two questions:

- Are you going to replace the `OSGroup` usages in dotnet/arcade with `([MSBuild]::IsOSPlatform('X'))`? We should not leak that custom property into Arcade.
- We should really rename the `OSGroup` property name. It still confuses me, `BuildOS` would really have been a better choice.

**Comment by Anipik** [comment]
> Are you going to replace the OSGroup usages in dotnet/arcade with ([MSBuild]::IsOSPlatform('X'))? We should not leak that custom property into Arcade.

No I am not changing the OS -> OSGroup as OSGroup is defined by runtime repo. However, I will remove OSGroup from the codeAnalysis part.

> We should really rename the OSGroup property name. It still confuses me, BuildOS would really have been a better choice.

File: src/libraries/Directory.Build.props (hunk 1 above)
Comment (Anipik) [reply src/libraries/Directory.Build.props]: Yes that's the final goal

**Comment by ViktorHofer** [comment]
> No I am not changing the OS -> OSGroup as OSGroup is defined by runtime repo.

I think you misunderstood me. I was asking if you remove the OSGroup usages not the OS ones. It seems that OSGroup conditions should be replaceable with the MSBuild IsOSPlatofmr functions.

**Comment by Anipik** [comment]
> I think you misunderstood me. I was asking if you remove the OSGroup usages not the OS ones. It seems that OSGroup conditions should be replaceable with the MSBuild IsOSPlatofmr functions.

yes we can do that in arcade.

**Comment by ViktorHofer** [comment]
> yes we can do that in arcade.

Created https://github.com/dotnet/arcade/issues/4907 to track this.

**Comment by Anipik** [comment]
@ViktorHofer BuildOS is causing some troubles. I am debugging it and we can make that change in the another pr.

@ViktorHofer @ericstj can you review this one ?

**Review by ericstj — APPROVED** [review]

**Comment by ericstj** [comment]
Keep an eye on the rolling build when merging.  Since this changes an entry-point property it's possible it has some differences between official/CI.

**Comment by Anipik** [comment]
sure I will keep an eye on those.

**Pull request merged**